src/odemis/acq/align/autofocus.py

```diff
@@ -501,7 +506,7 @@ def _DoExhaustiveFocus(future, detector, emt, focus, dfbkg, good_focus, rng_focu
         # we know that upper_bound is excluded but: 1. realistically the best focus
         # position would not be there 2. the upper_bound - orig_pos range is not
         # expected to be precisely a multiple of the step anyway
-        for next_pos in numpy.arange(orig_pos, upper_bound, step):
+        for next_pos in numpy.linspace(orig_pos, upper_bound, (upper_bound - orig_pos) / step, endpoint=False):
```

Comment:
You should update the comment about this new behaviour. That said, I think it's simpler if you include the end_point (upper_bound). The reason it wasn't included was because arange would typically skip it. Now, with linspace, it's easy to include it.